smart_open/tests/test_smart_open.py

```diff
@@ -882,17 +913,22 @@ def test_open_gz(self):
         assert m.hexdigest() == '18473e60f8c7c98d29d65bf805736a0d', \
             'Failed to read gzip'
 
-    def test_write_read_gz(self):
-        """Can write and read gzip?"""
-        with tempfile.NamedTemporaryFile('wb', suffix='.gz', delete=False) as infile:
+    def _test_write_read(self, suffix):
+        with tempfile.NamedTemporaryFile('wb', suffix=suffix, delete=False) as infile:
             test_file_name = infile.name
         self.write_read_assertion(test_file_name)
 
+    def test_write_read_gz(self):
+        """Can write and read gzip?"""
+        self._test_write_read('.gz')
+
     def test_write_read_bz2(self):
         """Can write and read bz2?"""
-        with tempfile.NamedTemporaryFile('wb', suffix='.bz2', delete=False) as infile:
-            test_file_name = infile.name
-        self.write_read_assertion(test_file_name)
+        self._test_write_read('.bz2')
+
+    def test_write_read_xz(self):
```

Comment:
You also need a test that successfully reads a known .xz file. Currently, you test that your .xz reader and writer are compatible with each other, but you're not testing whether they are actually working with the .xz format (e.g. they could be using gzip by mistake and this test would still pass).

Reply:
We don't test it for bzip2 as well (just noticing). OK, I will add this test.